src/Compilers/VisualBasic/Portable/Microsoft.CodeAnalysis.VisualBasic.vbproj

```diff
@@ -4,6 +4,7 @@
   <PropertyGroup>
     <OutputType>Library</OutputType>
     <TargetFrameworks>net6.0;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true'">net8.0;$(TargetFrameworks)</TargetFrameworks>
```

Comment:
Rather than put this condition in all these files I wonder if we should centralize a bit. Maybe in `Settings.props` have 

```xml
    <SourceBuildTargetFrameworks>net6.0</SourceBuildTargetFrameworks>
    <TargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true'">net8.0;net6.0</SourceBuildTargetFrameworks>
```

Then we can just use the `$(SourceBuildtargetFrameworks)` in the project files. The other advantage is we could further condition that we disable the `net8.0` when outside CI / official build. That way it doesn't build in the developer inner loop. 

I really don't like the name of the property though, open to pretty much any other suggestion 😄 

@333fred, @chsienki

Reply:
Well, see, I'm not sure whether I should be suggesting an alternative to `SourceBuildTargetFrameworks`, `TargetFrameworks`, or `SourceBuildtargetFrameworks` 😛

Reply:
Done. Btw, some projects packaged as part of source build target only `netstandard2.0`, they didn't have `net6.0`, so I didn't add `net8.0` to them, hope that's correct.